event.go

```diff
@@ -35,3 +35,26 @@ type DragEvent struct {
 	PointEvent
 	Dragged Delta
 }
+
+type EventFunc interface {
+	Execute()
+}
+
+type SimpleEventFunc func()
+
+func (fn SimpleEventFunc) Execute() {
+	fn()
+}
+
+type DragEventFunc struct {
+	wid Draggable
+	ev  *DragEvent
+}
+
+func NewDragEventFunc(wid Draggable, ev *DragEvent) *DragEventFunc {
+	return &DragEventFunc{wid, ev}
+}
+
+func (drag *DragEventFunc) Execute() {
+	drag.wid.Dragged(drag.ev)
+}
```

Comment:
I know this is a draft PR but just FYI: this adds a public API for something that I suppose only was intended to be used internally? You might want to use one of the packages in the `internal/` folder.